src/query/graphite/native/builtin_functions.go

```diff
@@ -200,6 +200,23 @@ func limit(_ *common.Context, series singlePathSpec, n int) (ts.SeriesList, erro
 	return r, nil
 }
 
+// grep takes a metric or a wildcard seriesList, followed by a regular
+// expression in double quotes. Excludes metrics that don’t match the regular expression.
+func grep(_ *common.Context, seriesList singlePathSpec, regex string) (ts.SeriesList, error) {
+	re := regexp.MustCompile(regex)
```

Comment:
This should not use `MustCompile` since the regex is user-defined. It should attempt to build the regex and return an error if it's invalid (mind adding a test for the error case as well?)

Reply:
good call